src/DynamoCoreWpf/ViewModels/Core/DynamoViewModelEventArgs.cs

```diff
@@ -153,4 +159,142 @@ public OpenDocumentationLinkEventArgs(Uri link)
+                InputDescriptions.Add(input.ToolTip);
+            }
+        }
+
+        static private string GetNamespace(NodeModel nodeModel, DynamoViewModel viewModel)
+        {
+            switch (nodeModel)
+            {
+                case Function function:
+                    var category = function.Category;
+                    var name = function.Name;
+                    if (CustomNodeHasCollisons(name, GetPackageName(nodeModel), viewModel))
+                    {
+                        var inputString = GetInputNames(function);
+                        return $"{category}.{name}({inputString})";
+                    }
+                    return $"{category}.{name}";
+
+                case DSFunctionBase dSFunction:
+                    var descriptor = dSFunction.Controller.Definition;
+                    var className = descriptor.ClassName;
+                    var functionName = descriptor.FunctionName;
+                    if (descriptor.IsOverloaded)
+                    {
+                        var inputString = GetInputNames(nodeModel);
+                        return $"{className}.{functionName}({inputString})";
+                    }
+
+                    return $"{className}.{functionName}";
+
+                case NodeModel node:
+                    var type = node.GetType();
+                    if (NodeModelHasCollisions(node.Name, viewModel))
+                    {
+                        return $"{type.FullName}({GetInputNames(nodeModel)})";
+                    }
+                    
+                    return type.FullName;
+                     
+
+                default:
+                    return string.Empty;
+            }
+        }
+
+        private static bool CustomNodeHasCollisons(string nodeName, string packageName, DynamoViewModel viewModel)
+        {
+            var pmExtension = viewModel.Model.GetPackageManagerExtension();
+            var package = pmExtension.PackageLoader.LocalPackages
+                .Where(x => x.Name == packageName)
+                .FirstOrDefault();
+
+            if (package is null)
+                return false;
+
+            var loadedNodesWithSameName =  package.LoadedCustomNodes
+                .Where(x => x.Name == nodeName)
+                .ToList();
+
+            if (loadedNodesWithSameName.Count == 1)
+                return false;
+            return true;
+        }
+
+        private static bool NodeModelHasCollisions(string nodeName, DynamoViewModel viewModel)
+        {     
+            var searchEntries = viewModel.Model.SearchModel.SearchEntries
+                .Where(x => x.Name == nodeName)
+                .Select(x => x).ToList();
+
+            if (searchEntries.Count() > 1)
+                return true;
+            return false;
+        }
+
+        private static string GetPackageName(NodeModel node)
+        {
+            return node.Category.Split(new char[] { '.' }).FirstOrDefault();
```

Comment:
This doesn't seem correct... what about if this is called on nodes that are not from a package?

Reply:
Is that actually possible? I use this to check if a custom node has any nodes with the same name in its package. Im simply extracting the first part of the category, i was under the impression that all custom nodes requires a category.

Maybe im wrong here?

Reply:
My point is that the name of this function might be confusing since it's called get package name, but thats not what it does.